### PR TITLE
Improve InputManager code quality

### PR DESCRIPTION
**Pull request by UselessToucan**
Various small improvements to the current `InputManager` code:

1. `Append` method does not modify the original collection. It returns new collection. `Append` call without field/variable assignment looks suspicious to me.
2. We do not need to create new lists on each input queue construction. We can reuse them.

### CODE REVIEW

File: osu.Framework/Input/InputManager.cs [hunk 1]
```diff
@@ -258,23 +264,11 @@ private void updateKeyRepeat(InputState state)
...
-            foreach (var h in InputHandlers)
-            {
-                var list = h.GetPendingInputs();
-                if (h.IsActive && h.Enabled)
-                    inputs.AddRange(list);
-            }
-
-            return inputs;
-        }
+        protected virtual List<IInput> GetPendingInputs() => InputHandlers.Where(h => h.IsActive && h.Enabled).SelectMany(h => h.GetPendingInputs()).ToList();
```
Comment (peppy) [inline osu.Framework/Input/InputManager.cs]: this is changing execution behaviour

File: osu.Framework/Input/InputManager.cs [hunk 2]
```diff
@@ -283,7 +277,7 @@ private IEnumerable<Drawable> buildInputQueue()
                 d.BuildKeyboardInputQueue(inputQueue);
 
             if (!unfocusIfNoLongerValid())
-                inputQueue.Append(FocusedDrawable);
```
Comment (peppy) [inline osu.Framework/Input/InputManager.cs]: given that this was working correctly without this, we should reassess if it is actually required.

File: osu.Framework/Input/InputManager.cs [hunk 3]
```diff
@@ -258,23 +264,11 @@ private void updateKeyRepeat(InputState state)
...
-            }
-
-            return inputs;
-        }
+        protected virtual List<IInput> GetPendingInputs() => InputHandlers.Where(h => h.IsActive && h.Enabled).SelectMany(h => h.GetPendingInputs()).ToList();
 
         private IEnumerable<Drawable> buildInputQueue()
         {
-            var inputQueue = new List<Drawable>();
+            inputQueue.Clear();
```
Comment (ekrctb) [inline osu.Framework/Input/InputManager.cs]: Potentially there is a reentrancy (a BuildKeyboardInputQueue retrives `GetContainingInputManager().InputQueue`) so I think it is safe to not optimize early.

Comment (UselessToucan) [reply osu.Framework/Input/InputManager.cs]: Don't understand where did you find this `InputQueue` call. Could you please provide a link to it?

Comment (ekrctb) [reply osu.Framework/Input/InputManager.cs]: I mean, a code could do that. No current code does that as far as I know.

File: osu.Framework/Input/InputManager.cs (hunk 1 above)
Comment (UselessToucan) [reply osu.Framework/Input/InputManager.cs]: Why is it necessary to call GetPendingInputs on disabled handlers? They shouldn't be subscribed to `host.Window` events i.e. they won't add anything to the `PendingInputs` queue.

Comment (ekrctb) [reply osu.Framework/Input/InputManager.cs]: Actually, ignoring inputs from disabled input handlers results in an incorrect behavior as I described in #1715.

**Comment by peppy** [comment]
Is this abandoned?

**Comment by UselessToucan** [comment]
Don't know what to do here since most of the changes turned out to be wrong. :(

**Comment by peppy** [comment]
I think it's fine to close this one for now. There's ongoing changes around these methods anyway (via #1713 amongst others).

**Pull request closed without merging**